pyop2/codegen/rep2loopy.py

```diff
@@ -562,7 +551,9 @@ def generate(builder, wrapper_name=None):
         wrapper = loopy.register_callable(
             wrapper,
             kernel.name,
-            PyOP2KernelCallable(name=kernel.name, access=tuple(builder.argument_accesses)))
+            PyOP2KernelCallable(name=kernel.name,
+                                arguments=context.function_args,
```

Comment:
Ah, at this point we don't have the instruction available, so we pass the whole dict. Is the name a good one?

Reply:
If you mean "can we key the dict using `kernel.name`? then the answer is yes and I've done that.

Reply:
I meant is `function_args` a good name for this dict?

Reply:
`kernel_parameters`?

Reply:
Better, I think.

Reply:
Done.